LiteCore/Database/Database.cc

```diff
@@ -234,10 +234,37 @@ namespace c4Internal {
         }
     }
 
+    unordered_set<string> Database::collectBlobs() {
+        RecordEnumerator::Options options;
+        options.contentOptions = kHasAttachmentsOnly;
+        RecordEnumerator e(defaultKeyStore(), 0, INT64_MAX, options);
+        unordered_set<string> usedDigests;
+        for(const Record& rec : e) {
+            auto doc = documentFactory().newDocumentInstance(rec);
+            doc->selectCurrentRevision();
+            do {
+                if(!doc->loadSelectedRevBodyIfAvailable()) {
+                    continue;
+                }
+                
+                const Dict* body = Value::fromTrustedData(doc->selectedRev.body)->asDict();
+                auto keys = _db->documentKeys();
+                Document::findBlobReferences(body, keys, [&usedDigests](const blobKey& key, uint64_t size) {
+                    usedDigests.insert(key.base64String());
```

Comment:
If you changed this to `key.filename()`, then BlobStore::deleteAllExcept wouldn't have to transform its filenames to look them up in the set, which would be a lot faster.